Makefile

```diff
@@ -53,14 +53,22 @@ include log/Makefile
 DEPS := $(patsubst %.o, %.c.d, $(OBJS))
 -include $(DEPS)
 
-$(PREFIX_PROG)phoenix-$(TARGET_FAMILY)-$(TARGET_SUBFAMILY).elf: $(OBJS)
+
+# By default ld adds program headers to first LOAD segment.
+# However, in kernel headers are not used and it is expected that kernel address space starts at _start.
+# This target creates linker script that do not add program headers into the first loadable segment.
+# 1) Get internal linker script with prefix and suffix containing noise.
+# 2) Match only internal linker script which is between two lines of =
+# 3) Remove those lines
+# 4) Remove "+ SIZEOF_HEADERS", which causes inclusion of program headers.
+$(PREFIX_O)$(TARGET_FAMILY)-$(TARGET_SUBFAMILY).ldt:
+	$(SIL)$(LD) $(LDFLAGS_PREFIX)--verbose 2>/dev/null | sed -n '/^=*$$/,/^=*$$/p' | sed '1,1d; $$d' | sed s/"\s*+\s*SIZEOF_HEADERS"// > "$@"
```

Comment:
`^=*$` will also match empty line as `*` means "zero or more characters". I'd change it to `^==*$` as there's no "one or more" in basic `sed` regexp and we also want to be compatible with MacOS.

Also you can grab the necessary lines in single command: `sed -n '/^==*$$/,/^==*$$/{/^=*$$/d;p;}'` - deletion can strip empty lines I guess.

Reply:
Thanks for suggestions! I've added `==*` however I feel that breaking the sed into two executions makes the script much more readable, thus I'd prefer to keep it as it is